In [426]:
import matplotlib
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import scipy.stats
import math

In [388]:
data = pd.read_csv('Cleaned_Data.csv')
data.head(5)

C:\Users\alvin\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (15,17,19,21,23,25,27,29,31,33) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Full Name  Last Name  Grad Year Position   Height   Height Source  \
0   Korey Foreman    Foreman       2021       DE    6' 4"    nike_opening   
1  J.T. Tuimoloau  Tuimoloau       2021       DE    6' 4"  national_preps   
2       JC Latham     Latham       2021       OT    6' 6"  national_preps   
3    Emeka Egbuka     Egbuka       2021       WR  6' 0.5"    nike_opening   
4   Dallas Turner     Turner       2021       DE  6' 2.5"     rivals_camp   

   Weight   Weight Source  Zcruit Rating Committed To  ...  Hand Size Source  \
0   248.0    nike_opening        96.6970          USC  ...               NaN   
1   275.0  national_preps        96.6970          NaN  ...               NaN   
2   279.0    nike_opening        96.6970      Alabama  ...               NaN   
3   190.0    nike_opening        96.3939   Ohio State  ...  national_combine   
4   223.0     rivals_camp        96.3939      Alabama  ...       rivals_camp   

  Powerball Toss Powerball Toss Source SPARQ Rating  SPARQ Rating Source  \
0           44.5          nike_opening        98.76         nike_opening   
1            NaN                   NaN          NaN                  NaN   
2            NaN                   NaN          NaN                  NaN   
3           41.0          nike_opening          NaN                  NaN   
4            NaN                   NaN          NaN                  NaN   

    GPA  ACT SAT  Unnamed: 37 Unnamed: 38  
0  3.03  NaN NaN          NaN         NaN  
1  3.00  NaN NaN          NaN         NaN  
2  2.35  NaN NaN          NaN         NaN  
3  4.00  NaN NaN          NaN         NaN  
4  3.70  NaN NaN          NaN         NaN  

[5 rows x 39 columns]

In [390]:
import re

vals = data["Height"].values
for element in vals:
    z = re.match("^(\d)'\s*(\d*\.*\d*)", element)
    if z:
        print((z.groups()))

('6', '4')
('6', '4')
('6', '6')
('6', '0.5')
('6', '2.5')
('6', '4.5')
('6', '5.5')
('6', '4')
('6', '1')
('6', '1')
('6', '4.5')
('6', '5.25')
('6', '5')
('6', '6')
('6', '2')
('6', '5')
('6', '2')
('6', '4')
('6', '8')
('6', '4')
('5', '10.75')
('6', '0.25')
('6', '0')
('5', '11')
('6', '4.5')
('6', '0')
('5', '11.75')
('6', '2')
('5', '10.5')
('6', '3.25')
('5', '11')
('6', '1')
('6', '4.75')
('5', '10')
('6', '5.5')
('6', '0')
('6', '5')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('5', '10')
('6', '3.5')
('5', '9.25')
('5', '11')
('6', '4')
('6', '3')
('5', '9.5')
('6', '3')
('6', '4.5')
('5', '7')
('6', '0.5')
('6', '5.5')
('6', '3')
('6', '0')
('6', '2')
('6', '1')
('6', '4')
('6', '5')
('6', '1')
('6', '2')
('6', '5')
('6', '3')
('6', '0.5')
('6', '1')
('6', '3')
('6', '3.5')
('5', '9')
('6', '2.25')
('6', '0')
('6', '3.75')
('6', '3.5')
('6', '0')
('5', '10')
('6', '2')
('6', '0')
('5', '11')
('6', '1')
('6', '2')
('6', '5')
('6', '2')
('6', '0')
('6', '3')
('6', '

('6', '6')
('6', '0')
('6', '6')
('6', '0')
('5', '10')
('6', '0')
('6', '2')
('6', '4')
('6', '0')
('6', '1')
('6', '4')
('6', '0')
('5', '9')
('6', '2')
('6', '5')
('6', '1')
('6', '4')
('6', '5')
('6', '2')
('5', '10')
('6', '3')
('5', '7.5')
('6', '2')
('6', '1')
('6', '2')
('6', '4')
('6', '4')
('6', '1')
('6', '0')
('6', '2')
('6', '0')
('6', '1')
('6', '1')
('5', '9.5')
('5', '10')
('5', '10.5')
('6', '1')
('5', '11')
('6', '0')
('6', '1')
('6', '3')
('5', '8.5')
('5', '10')
('6', '0.25')
('5', '9.75')
('6', '0.5')
('6', '3')
('6', '1.5')
('6', '8')
('6', '3')
('6', '1.75')
('6', '2')
('6', '3.5')
('6', '0')
('6', '0')
('6', '0')
('5', '11')
('6', '3')
('6', '4')
('6', '1')
('6', '0')
('6', '0')
('5', '10')
('6', '0')
('5', '10')
('5', '11')
('5', '11')
('6', '3')
('6', '6')
('6', '1')
('5', '11')
('6', '3')
('6', '3')
('5', '9')
('6', '2')
('6', '3.5')
('6', '2')
('5', '8')
('5', '11')
('5', '11')
('6', '0')
('5', '11.5')
('5', '10')
('6', '5')
('5', '10')
('6', '2')
('5', '10.

('6', '0')
('5', '7')
('5', '11')
('5', '10')
('5', '10')
('6', '5')
('6', '0')
('5', '11')
('5', '11')
('6', '0')
('6', '3')
('5', '9')
('5', '11')
('5', '4')
('5', '6')
('5', '11')
('5', '5')
('6', '0')
('5', '11')
('5', '6')
('5', '7')
('5', '3')
('5', '11')
('5', '8')
('6', '0')
('5', '5')
('5', '10')
('5', '10')
('5', '10')
('6', '2')
('6', '1.5')
('5', '5')
('5', '8')
('6', '4')
('5', '7')
('5', '9')
('5', '8')
('6', '0.5')
('6', '4')
('6', '4')
('5', '10.5')
('6', '2')
('6', '1')
('6', '1')
('5', '11')
('5', '6')
('5', '9')
('6', '1')
('5', '5')
('6', '0')
('5', '5.5')
('5', '10')
('5', '8')
('5', '7')
('5', '8')
('5', '10')
('5', '9')
('5', '10')
('6', '0.25')
('5', '10')
('5', '8')
('5', '11')
('5', '11')
('5', '6')
('6', '0')
('5', '11')
('5', '10')
('5', '6')
('5', '11')
('6', '3')
('5', '6')
('5', '4')
('5', '8.5')
('5', '7')
('5', '9')
('6', '1')
('5', '11')
('5', '9')
('6', '0')
('5', '9')
('5', '11')
('6', '2')
('6', '1.25')
('5', '11')
('5', '9.75')
('5', '10')
('5', '5

('6', '1')
('6', '4')
('6', '4')
('6', '1')
('6', '1')
('6', '3')
('6', '5')
('6', '4')
('6', '2')
('6', '4')
('5', '11')
('6', '3')
('6', '2')
('6', '1')
('6', '0')
('6', '0')
('6', '3')
('6', '3')
('6', '0')
('6', '0')
('6', '0')
('5', '11')
('5', '10')
('6', '2')
('6', '1')
('5', '10')
('6', '0')
('5', '9')
('6', '1')
('6', '0')
('5', '10')
('5', '11')
('6', '1')
('6', '3')
('6', '1')
('6', '2')
('6', '0')
('6', '2')
('6', '2')
('6', '2')
('6', '3')
('6', '4')
('6', '0')
('6', '5')
('5', '10')
('6', '2')
('6', '0')
('6', '3')
('5', '9')
('6', '0')
('5', '11')
('6', '3')
('5', '9')
('6', '2')
('6', '0')
('6', '1')
('6', '1')
('5', '8')
('6', '1')
('6', '2')
('6', '0')
('6', '3')
('5', '10')
('6', '1')
('5', '9')
('6', '2')
('5', '10')
('6', '1')
('6', '0')
('5', '10')
('6', '0')
('5', '9')
('6', '2')
('5', '10')
('6', '1')
('6', '3')
('6', '2')
('6', '0')
('5', '9')
('5', '11')
('5', '10')
('6', '4')
('6', '3')
('6', '3')
('6', '2')
('6', '0')
('6', '3')
('6', '0')
('6', '4')
('5', '

('5', '10')
('6', '1')
('5', '10')
('6', '2')
('6', '2')
('6', '0')
('6', '7')
('6', '3')
('6', '1')
('6', '3')
('5', '8')
('5', '10')
('6', '0')
('6', '3')
('6', '2')
('6', '1')
('6', '2')
('6', '4')
('6', '3')
('6', '2')
('6', '3')
('6', '3')
('6', '0')
('6', '6')
('6', '5')
('6', '3')
('6', '3')
('6', '3')
('6', '3')
('6', '2')
('6', '5')
('6', '1')
('6', '3')
('6', '1')
('6', '4')
('6', '2')
('6', '3')
('6', '3')
('6', '4')
('6', '3')
('6', '0')
('6', '5')
('6', '4')
('6', '5')
('6', '4')
('6', '6')
('6', '7')
('6', '5')
('6', '3')
('6', '1')
('6', '2')
('6', '3')
('6', '5')
('6', '4')
('6', '6')
('6', '4')
('6', '3')
('6', '3')
('6', '3')
('6', '2')
('6', '5')
('6', '2')
('6', '7')
('6', '6')
('6', '2')
('6', '1')
('6', '0')
('6', '5')
('6', '6')
('6', '4')
('6', '4')
('6', '3')
('6', '3')
('6', '2')
('6', '2')
('6', '0')
('6', '6')
('6', '4')
('6', '7')
('6', '5')
('6', '4')
('6', '3')
('6', '3')
('6', '6')
('6', '4')
('6', '3')
('6', '3')
('6', '2')
('6', '2')
('6', '1')
('6', '

('6', '4')
('6', '0')
('6', '1')
('6', '1')
('5', '9.5')
('6', '1.25')
('6', '0')
('5', '11')
('5', '11')
('6', '3')
('5', '10')
('6', '3')
('6', '0')
('5', '11')
('5', '10')
('6', '3')
('5', '6.5')
('6', '0')
('5', '10')
('6', '3')
('6', '3')
('6', '1')
('6', '0')
('5', '6')
('5', '5')
('5', '10')
('5', '11')
('6', '0')
('5', '9')
('6', '0')
('5', '9')
('5', '11')
('5', '10')
('6', '2')
('6', '0')
('6', '1')
('5', '8')
('5', '7.5')
('5', '10')
('5', '11')
('5', '10')
('6', '2')
('5', '9')
('6', '1')
('5', '8')
('5', '4')
('5', '8')
('5', '9')
('5', '3')
('5', '10')
('5', '6')
('5', '6')
('5', '10')
('5', '9')
('5', '8')
('5', '10')
('6', '2')
('5', '10')
('5', '8')
('5', '10')
('5', '10')
('5', '8')
('5', '6.75')
('5', '6')
('6', '0')
('5', '11')
('6', '1')
('5', '11')
('5', '7')
('5', '8')
('5', '8')
('6', '1')
('5', '9')
('6', '4')
('6', '2')
('5', '8')
('5', '10')
('5', '11')
('6', '1')
('5', '10')
('5', '10')
('5', '9')
('5', '7')
('5', '11')
('5', '7')
('5', '11')
('5', '8')
('5'

('5', '9')
('5', '9')
('5', '10')
('5', '10')
('6', '1')
('6', '2')
('6', '1')
('6', '0')
('6', '0')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('5', '11')
('6', '5')
('6', '0')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '11')
('6', '1')
('6', '3')
('6', '2')
('6', '2')
('6', '1')
('6', '0')
('6', '0')
('5', '9')
('5', '7')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '10')
('6', '3')
('6', '0')
('6', '0')
('5', '11')
('5', '10')
('5', '10')
('5', '9')
('5', '11')
('6', '3')
('6', '2')
('6', '2')
('6', '1')
('5', '11')
('5', '11')
('5', '10')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('5', '7')
('5', '9')
('5', '8')
('5', '7')
('5', '10')
('5', '10')
('6', '0')
('5', '9')
('5', '9')
('5', '8')
('5', '7')
('5', '11')
('6', '2')
('5', '10')
('5', '7')
('5', '7')
('5', '10')
('6', '1')
('5', '9')
('6', '7')
('6', '0')
('5', '8')
('6', '2')
('6', '4')
('5', '11')
('5', '9')
('5', '11')
('5', '9')
('6', '1')
('5', '10')
('6', '3')
(

('6', '0')
('5', '8')
('5', '8')
('5', '6')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('6', '0')
('5', '10')
('6', '4')
('6', '4')
('6', '4')
('6', '4')
('6', '3')
('6', '2')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('5', '9')
('5', '9')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('6', '7')
('5', '10')
('6', '5')
('6', '4')
('6', '4')
('6', '3')
('6', '3')
('6', '2')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('6', '0')
('5', '9')
('5', '11')
('5', '10')
('5', '11')
('6', '2')
('6', '0')
('5', '10')
('5', '10')
('5', '9')
('6', '4')
('6', '4')
('6', '4')
('6', '3')
('6', '3')
('6', '3')
('6', '3')
('6', '3')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', 

('6', '0')
('6', '0')
('6', '0')
('5', '6')
('5', '11')
('6', '2')
('5', '7')
('5', '8')
('5', '9')
('6', '1')
('6', '1')
('6', '0')
('5', '8')
('5', '10')
('6', '2')
('5', '10')
('5', '8')
('6', '2')
('5', '9')
('5', '7')
('5', '10')
('5', '7')
('5', '9')
('5', '11')
('5', '8')
('5', '3')
('5', '10')
('5', '8')
('5', '8')
('6', '1')
('6', '0')
('6', '0')
('5', '10')
('5', '7')
('5', '8')
('5', '11')
('6', '4')
('6', '0')
('5', '11')
('5', '8')
('5', '7')
('5', '10')
('5', '9')
('5', '11')
('6', '2.5')
('6', '3.5')
('6', '2')
('6', '3')
('5', '7.5')
('6', '2.5')
('5', '8')
('5', '10')
('5', '9')
('6', '3')
('5', '8')
('6', '4')
('5', '8')
('5', '7')
('6', '1')
('5', '11')
('5', '9')
('6', '2')
('5', '8')
('5', '11')
('5', '9')
('5', '9')
('5', '8')
('5', '9')
('6', '0')
('5', '11')
('5', '8')
('5', '7')
('5', '11')
('5', '8')
('6', '1')
('6', '1')
('5', '11')
('6', '1')
('6', '0')
('6', '0')
('5', '5')
('5', '10')
('6', '1')
('5', '10')
('5', '10')
('5', '11')
('5', '8')
('5', '6')
('5

('5', '11')
('6', '1')
('5', '9')
('5', '10')
('6', '0')
('5', '8')
('5', '11')
('5', '8')
('5', '8')
('5', '10')
('5', '11')
('5', '4')
('5', '10')
('6', '0')
('5', '9')
('5', '11')
('5', '8')
('5', '7')
('5', '9')
('5', '8')
('5', '10')
('5', '9')
('5', '9')
('5', '11')
('6', '1')
('6', '2')
('5', '11')
('6', '0')
('5', '11')
('5', '11')
('5', '7')
('6', '4')
('6', '0')
('6', '1')
('6', '2')
('5', '11')
('5', '9')
('6', '3')
('6', '2')
('6', '2')
('6', '4')
('5', '11')
('5', '10')
('5', '11')
('6', '1')
('5', '10')
('6', '1')
('6', '0')
('5', '9')
('6', '0')
('6', '2')
('5', '9')
('6', '0')
('6', '2')
('6', '5')
('6', '1')
('5', '10')
('6', '0')
('5', '11')
('5', '10')
('5', '10')
('6', '0')
('6', '0')
('6', '0')
('6', '1')
('5', '11')
('6', '1')
('6', '1')
('5', '11')
('6', '3')
('6', '1')
('6', '1')
('5', '11')
('5', '11')
('6', '1')
('6', '0')
('6', '4')
('6', '0')
('6', '3')
('6', '3')
('5', '10')
('5', '10')
('6', '1')
('5', '9')
('5', '10')
('6', '3')
('5', '10')
('6', '0')
('6

('5', '5')
('5', '9')
('5', '5')
('5', '8')
('5', '9')
('5', '9')
('5', '9')
('5', '8')
('6', '0')
('5', '5')
('5', '8')
('5', '10')
('5', '9')
('6', '3')
('6', '0')
('5', '10')
('5', '11')
('6', '2')
('5', '9')
('5', '11')
('5', '5')
('5', '11')
('5', '9')
('5', '7')
('5', '10')
('5', '7')
('6', '2')
('5', '10')
('5', '11')
('6', '0')
('5', '10')
('5', '6')
('5', '10')
('6', '0')
('5', '10')
('5', '10')
('5', '6')
('5', '11')
('5', '9')
('6', '1')
('5', '11')
('6', '0')
('5', '10')
('5', '7')
('5', '11')
('5', '10')
('6', '0')
('5', '10')
('5', '11')
('6', '0')
('5', '10')
('5', '10')
('6', '2')
('6', '2')
('6', '3')
('6', '2')
('6', '3')
('6', '0')
('5', '10')
('5', '10')
('5', '9')
('6', '3')
('5', '10')
('6', '4')
('5', '11')
('5', '11')
('6', '1')
('5', '10')
('6', '4')
('6', '2')
('5', '10')
('6', '0')
('5', '7')
('6', '1')
('6', '3')
('6', '1')
('5', '9')
('5', '10')
('5', '9')
('5', '11')
('5', '7')
('5', '11')
('6', '3')
('6', '0')
('5', '11')
('6', '4')
('6', '3')
('6', '1')


('5', '10')
('5', '11')
('6', '1')
('5', '9')
('5', '11')
('5', '9')
('5', '11')
('5', '10')
('5', '10')
('6', '0')
('6', '0')
('5', '10')
('5', '11')
('5', '11')
('5', '11')
('5', '8')
('5', '6')
('5', '9')
('5', '10')
('6', '0')
('6', '3')
('6', '0')
('5', '9')
('5', '11')
('5', '6')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('6', '5')
('6', '4')
('6', '0')
('6', '2')
('5', '11')
('6', '1')
('6', '1')
('5', '10')
('6', '0')
('6', '1')
('5', '8')
('5', '6')
('5', '11')
('6', '2')
('6', '0')
('6', '0')
('5', '10')
('5', '11')
('5', '9')
('6', '0')
('6', '0')
('5', '11')
('5', '11')
('5', '9')
('5', '10')
('5', '9')
('6', '1')
('6', '2')
('6', '1')
('5', '9')
('5', '9')
('6', '2')
('5', '5')
('5', '11')
('5', '10')
('6', '2')
('6', '2')
('5', '8')
('5', '10')
('5', '10')
('5', '11')
('6', '0')
('6', '0')
('6', '2')
('5', '11')
('5', '8')
('5', '9')
('5', '9')
('5', '11')
('5', '10')
('5', '10')
('5', '8')
('6', '1')
('5', '8')
('5', '9')
('5', '10')
('5', '5')
('5', '8')
('5', '11

('5', '7')
('5', '5')
('6', '0')
('5', '6')
('6', '0')
('5', '7')
('5', '10')
('6', '1')
('5', '6')
('5', '9')
('5', '9')
('5', '8')
('5', '10')
('5', '11')
('5', '11')
('5', '11')
('5', '8.5')
('5', '10')
('6', '0')
('6', '0')
('6', '0')
('5', '7')
('5', '10')
('6', '2')
('6', '0')
('5', '8')
('5', '8')
('5', '7')
('5', '10')
('5', '6')
('5', '9')
('5', '4')
('5', '6')
('5', '10')
('6', '1')
('6', '7')
('5', '6')
('5', '7')
('6', '1')
('5', '11')
('5', '8')
('5', '8')
('5', '5')
('5', '7')
('5', '10')
('6', '1')
('6', '2')
('5', '10')
('5', '10')
('5', '9')
('5', '11')
('5', '9')
('6', '4')
('5', '11')
('5', '7')
('5', '8')
('5', '11')
('5', '8')
('5', '8')
('6', '4')
('5', '11')
('5', '10')
('5', '11')
('6', '1')
('6', '0')
('5', '11')
('5', '9')
('5', '11')
('5', '11')
('6', '0')
('6', '2')
('5', '9')
('6', '4')
('5', '9')
('5', '10')
('5', '10')
('6', '0')
('6', '1')
('5', '10')
('5', '11')
('6', '1')
('6', '0')
('5', '8')
('6', '2')
('5', '8')
('6', '3')
('6', '2')
('6', '0')
('5'

('6', '2')
('6', '1')
('5', '8')
('5', '10')
('5', '9')
('6', '1')
('5', '7')
('5', '9')
('5', '7')
('5', '9')
('5', '10')
('6', '4')
('5', '11')
('5', '8')
('6', '1')
('5', '8')
('6', '0')
('6', '2')
('6', '0')
('6', '0')
('6', '0')
('5', '10')
('5', '6')
('5', '10')
('6', '0')
('5', '10')
('5', '6')
('6', '3')
('5', '11')
('6', '0')
('5', '10')
('6', '0')
('5', '8')
('5', '7')
('6', '0')
('5', '8')
('6', '2')
('5', '8')
('6', '2')
('5', '11')
('5', '10')
('5', '8')
('5', '6')
('6', '0')
('5', '7')
('5', '7')
('5', '9')
('5', '9')
('5', '7')
('5', '10')
('5', '6')
('5', '10')
('5', '5')
('5', '8')
('5', '8')
('5', '4')
('6', '1')
('5', '4')
('5', '10')
('5', '6')
('6', '0')
('5', '11')
('5', '9')
('5', '5')
('6', '0')
('5', '6')
('5', '9')
('6', '1')
('5', '5')
('5', '6')
('6', '1')
('5', '8')
('5', '6')
('5', '7')
('5', '9')
('5', '10')
('5', '5')
('6', '0')
('5', '8')
('5', '7')
('5', '10')
('5', '8')
('5', '8')
('5', '10')
('5', '10')
('5', '11')
('5', '9')
('5', '9')
('6', '0')
('

('5', '7')
('5', '7')
('5', '6')
('5', '4')
('5', '9.75')
('5', '9')
('5', '9')
('5', '5')
('5', '8')
('5', '7')
('5', '6')
('5', '11')
('6', '0')
('5', '2')
('5', '6')
('5', '3')
('5', '11')
('6', '3')
('5', '1')
('6', '0')
('5', '8')
('5', '6')
('5', '11')
('5', '9')
('5', '8')
('5', '8')
('5', '4')
('5', '8')
('5', '9')
('5', '9')
('6', '0')
('5', '7')
('5', '7')
('5', '4')
('6', '2')
('5', '6')
('5', '4')
('5', '7')
('6', '1')
('5', '6')
('5', '4')
('5', '9')
('5', '9')
('5', '7')
('6', '0')
('5', '11')
('6', '4')
('5', '8')
('5', '5')
('5', '9')
('5', '3')
('5', '10')
('6', '0')
('5', '1')
('5', '8')
('5', '8')
('5', '11')
('5', '10')
('6', '1')
('5', '8')
('5', '10')
('5', '8')
('5', '11')
('6', '0')
('5', '4')
('5', '8')
('5', '9')
('5', '7')
('5', '4')
('5', '4')
('5', '8')
('5', '10')
('5', '11')
('5', '7')
('5', '11')
('5', '9')
('5', '10')
('5', '3')
('5', '8')
('6', '2')
('6', '0')
('6', '4')
('5', '9')
('5', '11')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '7')
('5'

('5', '8')
('5', '6')
('5', '1')
('5', '7')
('5', '9')
('5', '8')
('5', '9')
('5', '7')
('5', '10')
('5', '5')
('5', '9')
('5', '10')
('6', '1')
('6', '0')
('5', '7')
('5', '11')
('5', '4')
('5', '9')
('5', '7')
('6', '3')
('5', '9')
('5', '4')
('6', '1')
('5', '5')
('5', '9')
('5', '7')
('5', '9')
('5', '7')
('5', '8')
('6', '2')
('5', '7')
('5', '7')
('6', '0')
('5', '4')
('5', '9')
('5', '10')
('6', '1')
('6', '0')
('6', '0')
('5', '5')
('6', '0')
('6', '0')
('5', '7')
('6', '1')
('6', '1')
('5', '3')
('5', '8')
('5', '7')
('5', '10')
('5', '7')
('5', '9')
('5', '6')
('6', '0')
('5', '5')
('5', '9')
('5', '11')
('5', '7.5')
('6', '1')
('5', '8')
('5', '8')
('6', '0')
('5', '11.25')
('6', '1.5')
('5', '11')
('5', '7')
('5', '11')
('5', '11')
('6', '1')
('5', '8')
('5', '10')
('5', '8')
('6', '3')
('5', '8')
('5', '5')
('6', '1')
('5', '9')
('6', '1')
('5', '10')
('5', '7')
('5', '4')
('6', '1')
('5', '10')
('5', '7')
('5', '9')
('5', '9')
('5', '10')
('5', '5')
('5', '9')
('5', '6')


('5', '10')
('5', '7')
('5', '5')
('5', '11')
('6', '3')
('5', '7')
('5', '9')
('5', '8')
('5', '6')
('5', '7')
('5', '11')
('6', '1')
('5', '6')
('5', '11')
('5', '6')
('5', '10')
('5', '6')
('5', '11')
('5', '8')
('5', '11')
('5', '10')
('5', '9')
('5', '7')
('6', '2')
('6', '0')
('5', '9')
('5', '3')
('5', '8')
('5', '8')
('5', '10')
('6', '2')
('5', '7')
('5', '8')
('5', '4')
('5', '10')
('5', '11')
('6', '0')
('6', '0')
('5', '4')
('5', '8')
('5', '10')
('5', '6')
('5', '8')
('5', '8')
('5', '2')
('6', '0')
('5', '5')
('6', '1')
('5', '4')
('6', '0')
('5', '7')
('5', '7')
('5', '5')
('5', '9')
('5', '10')
('5', '9')
('5', '6')
('5', '8')
('5', '4.5')
('5', '4')
('5', '4')
('5', '10')
('5', '7')
('5', '9')
('6', '2')
('6', '0')
('6', '1')
('6', '1')
('5', '4')
('5', '6')
('5', '5')
('5', '6')
('6', '0')
('6', '0')
('5', '11')
('5', '8')
('6', '1')
('6', '1')
('6', '0')
('5', '10')
('5', '6')
('6', '2')
('5', '7')
('5', '9')
('5', '8')
('5', '7')
('5', '10')
('5', '9')
('5', '9')
('

('5', '6')
('6', '1')
('5', '4')
('5', '6')
('5', '6')
('6', '5.25')
('5', '10')
('6', '0')
('5', '11')
('5', '6')
('5', '9')
('5', '10')
('5', '10')
('5', '7')
('6', '3')
('5', '9')
('5', '7')
('6', '2')
('6', '0')
('5', '9')
('5', '11')
('5', '6')
('6', '1')
('6', '1')
('6', '3')
('5', '6')
('6', '2')
('5', '9')
('6', '6')
('5', '3')
('5', '10')
('5', '11')
('5', '10')
('5', '6')
('5', '9')
('5', '9')
('5', '2')
('6', '0')
('5', '7')
('5', '6')
('5', '6')
('5', '10')
('5', '8')
('5', '10')
('5', '7')
('5', '5')
('6', '0')
('6', '0')
('5', '2')
('6', '2')
('5', '10')
('6', '2')
('5', '4')
('5', '6')
('5', '10')
('5', '11')
('5', '6')
('5', '7')
('5', '10')
('6', '0')
('5', '11')
('6', '1')
('6', '1')
('6', '3')
('6', '4')
('6', '1')
('5', '9')
('5', '6')
('5', '10')
('5', '11')
('6', '3')
('5', '11')
('6', '3')
('6', '0')
('6', '1')
('5', '10')
('5', '10')
('5', '4')
('5', '11')
('5', '11')
('5', '8')
('5', '9')
('5', '9')
('5', '8')
('5', '7')
('6', '2')
('5', '9')
('5', '9')
('5', '

('6', '2')
('5', '7')
('5', '8')
('6', '1')
('5', '6')
('5', '10')
('5', '7')
('5', '9')
('5', '9')
('6', '3.5')
('5', '8')
('5', '10')
('5', '8')
('5', '2')
('6', '4.5')
('5', '11')
('6', '3')
('5', '11')
('5', '9')
('5', '10')
('5', '8')
('5', '11')
('5', '11')
('5', '3')
('6', '0')
('5', '9')
('5', '9')
('5', '8')
('6', '1')
('5', '11')
('5', '9')
('5', '10')
('5', '10')
('5', '11.5')
('5', '6')
('6', '5')
('6', '1')
('5', '11')
('6', '9.25')
('5', '11')
('5', '10')
('5', '9')
('5', '8')
('6', '1')
('5', '8')
('6', '3')
('5', '1')
('5', '6')
('5', '4')
('5', '10')
('5', '11')
('5', '9')
('5', '11')
('6', '1')
('6', '0')
('5', '7')
('6', '0')
('5', '11')
('6', '1')
('5', '7')
('5', '5')
('6', '0')
('5', '9')
('5', '9')
('5', '9')
('5', '5')
('6', '2')
('6', '4')
('5', '8')
('5', '11')
('5', '8')
('5', '6')
('5', '10')
('5', '8')
('5', '11')
('5', '9')
('6', '2')
('6', '3')
('5', '9')
('5', '11')
('5', '11')
('5', '6')
('6', '0')
('5', '6')
('5', '3')
('5', '3')
('5', '7')
('6', '2')


('5', '11')
('5', '11')
('6', '1')
('6', '2')
('5', '10')
('6', '1')
('6', '3')
('5', '10')
('5', '11')
('5', '10')
('6', '2')
('5', '11')
('6', '1')
('6', '1')
('6', '2')
('6', '3')
('6', '0')
('6', '4')
('6', '1')
('6', '1')
('5', '9')
('6', '2')
('6', '6')
('6', '1')
('6', '1')
('6', '2')
('5', '7')
('5', '11')
('6', '0')
('5', '9')
('5', '11')
('6', '2')
('6', '2')
('5', '11')
('6', '1')
('6', '2')
('6', '3')
('5', '10')
('5', '8')
('6', '4')
('6', '4')
('6', '4')
('6', '2')
('6', '0')
('5', '11')
('6', '4')
('6', '0')
('6', '0')
('5', '10')
('6', '3')
('6', '3')
('6', '4')
('6', '4')
('6', '4')
('6', '2')
('5', '10')
('6', '1')
('6', '0')
('5', '11')
('5', '10')
('6', '3')
('5', '10')
('6', '1')
('5', '11')
('5', '10')
('6', '0')
('6', '0')
('6', '1')
('5', '11')
('6', '2')
('6', '2')
('5', '11')
('6', '1')
('6', '0')
('6', '1')
('6', '0')
('5', '11')
('5', '11')
('6', '3')
('5', '9')
('6', '0')
('6', '0')
('6', '5')
('5', '10')
('6', '3')
('6', '1')
('6', '2')
('6', '2')
('6', '1

('6', '1')
('6', '0')
('5', '10')
('5', '11')
('6', '3')
('5', '10')
('6', '3')
('5', '6')
('5', '10')
('5', '9')
('6', '1')
('5', '6')
('5', '7')
('5', '8')
('5', '5')
('6', '6')
('6', '3')
('5', '11')
('5', '8')
('6', '0')
('6', '0')
('5', '10')
('5', '5')
('6', '5')
('6', '3')
('5', '11')
('6', '0')
('5', '9')
('6', '1')
('6', '1')
('5', '9')
('5', '7')
('6', '0')
('5', '10')
('6', '0')
('6', '1')
('5', '9')
('5', '10')
('5', '8')
('6', '0')
('5', '8')
('5', '10')
('5', '11')
('6', '0')
('6', '0')
('6', '1')
('5', '9')
('5', '11')
('5', '10')
('5', '10')
('6', '0')
('6', '0')
('6', '4')
('6', '2')
('5', '9')
('5', '6')
('5', '7')
('6', '2')
('5', '9')
('6', '1')
('5', '8')
('5', '10')
('6', '1')
('6', '1')
('5', '11')
('6', '3')
('6', '2')
('6', '0')
('6', '1')
('5', '9')
('6', '1')
('6', '5')
('5', '9')
('5', '10')
('5', '9')
('5', '11')
('5', '8')
('6', '1')
('6', '2')
('6', '0')
('5', '10')
('5', '6')
('5', '11')
('5', '10')
('6', '0')
('5', '6')
('5', '9')
('6', '0')
('5', '8')


('5', '9')
('5', '8')
('5', '8')
('5', '6')
('5', '5')
('5', '11')
('5', '10')
('5', '3')
('5', '8')
('5', '9')
('5', '8')
('5', '8')
('5', '7')
('5', '7')
('5', '6')
('5', '3')
('5', '5')
('5', '7')
('5', '7')
('5', '6')
('5', '5')
('5', '6')
('5', '4')
('5', '9')
('5', '0')
('6', '2')
('5', '10')
('5', '11')
('5', '9')
('6', '1')
('6', '2')
('5', '11')
('6', '1')
('5', '9')
('5', '9')
('6', '2')
('5', '8')
('6', '1')
('5', '8')
('5', '10')
('6', '3')
('6', '3')
('6', '0')
('6', '2')
('6', '0')
('6', '4')
('6', '4')
('6', '5')
('6', '1')
('5', '11')
('6', '2')
('6', '5')
('6', '2')
('5', '8')
('6', '0')
('5', '10')
('6', '2')
('5', '9')
('5', '11')
('6', '0')
('5', '10')
('6', '0')
('5', '8')
('6', '1')
('5', '11')
('6', '3')
('6', '1')
('5', '9')
('5', '10')
('6', '3')
('6', '0')
('6', '5')
('6', '2')
('6', '4')
('6', '3')
('5', '11')
('6', '1')
('5', '9')
('6', '3')
('6', '1')
('6', '1')
('6', '1')
('5', '8')
('5', '9')
('5', '7')
('5', '11')
('5', '11')
('5', '9')
('6', '2')
('5', 

('6', '1')
('6', '1')
('6', '3')
('6', '3')
('5', '10')
('5', '5')
('5', '3')
('6', '2')
('5', '11')
('6', '6')
('6', '4')
('6', '6')
('6', '4')
('5', '11')
('6', '0')
('5', '9')
('5', '9')
('5', '10')
('6', '3')
('6', '2')
('6', '0')
('5', '7')
('5', '8')
('6', '1')
('6', '2')
('5', '9')
('5', '10')
('6', '0')
('5', '11')
('5', '11')
('5', '8')
('6', '0')
('5', '11')
('6', '0')
('6', '0')
('6', '0')
('5', '8')
('5', '11')
('6', '1')
('5', '11')
('5', '10')
('5', '6')
('6', '0')
('6', '1')
('5', '10')
('6', '1')
('6', '3')
('5', '11')
('5', '9')
('6', '1')
('6', '0')
('5', '10')
('6', '2')
('6', '3')
('5', '9')
('6', '2')
('5', '7')
('5', '8')
('5', '11')
('5', '7')
('6', '0')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('5', '11')
('5', '9')
('5', '10')
('5', '8')
('5', '9')
('6', '2')
('5', '10')
('6', '3')
('6', '2')
('5', '11')
('5', '9')
('5', '11')
('5', '10')
('6', '0')
('6', '3')
('6', '2')
('6', '1')
('5', '11')
('5', '11')
('6', '1')
('5', '8')
('5', '7')
('5', '11')
('5', '1

('6', '0')
('5', '11')
('6', '1')
('6', '1')
('6', '1')
('6', '2')
('6', '2')
('5', '6')
('5', '9')
('6', '0')
('5', '10')
('5', '11')
('6', '3')
('5', '10')
('5', '7')
('6', '1')
('5', '8')
('5', '11')
('6', '1')
('6', '1')
('6', '6')
('6', '3')
('6', '1')
('5', '10')
('6', '4')
('6', '6')
('5', '11')
('6', '1')
('6', '2')
('6', '1')
('6', '3')
('6', '3')
('5', '7')
('5', '11')
('5', '11')
('6', '1')
('6', '3')
('6', '1')
('6', '1')
('5', '11')
('6', '9')
('6', '1')
('6', '0')
('6', '0')
('5', '9')
('5', '7')
('5', '11')
('6', '1')
('5', '11')
('5', '11')
('6', '1')
('5', '11')
('5', '10')
('5', '7')
('5', '6')
('5', '8')
('5', '6')
('5', '7')
('6', '0')
('6', '0')
('5', '8')
('6', '0')
('5', '7')
('5', '10')
('6', '3')
('6', '0')
('6', '0')
('5', '11')
('5', '11')
('6', '2')
('6', '0')
('6', '0')
('6', '0')
('5', '6')
('5', '9')
('6', '0')
('6', '4')
('5', '11')
('5', '8')
('6', '3')
('6', '1')
('6', '1')
('6', '2')
('5', '11')
('5', '10')
('6', '1')
('5', '8')
('5', '8')
('6', '0')


('5', '10')
('6', '0')
('5', '8')
('5', '8')
('5', '7')
('6', '0')
('5', '4')
('6', '0')
('5', '3')
('5', '10')
('5', '10')
('5', '9')
('5', '11')
('5', '11')
('5', '9')
('5', '5')
('5', '11')
('6', '0')
('5', '2')
('6', '1')
('6', '1')
('5', '9')
('5', '11')
('6', '2')
('6', '1')
('5', '6')
('6', '2')
('6', '0')
('6', '3')
('5', '10')
('5', '8')
('5', '9')
('5', '7')
('5', '7')
('5', '7')
('6', '0')
('5', '8')
('6', '1')
('5', '9')
('5', '10')
('6', '2')
('5', '10')
('5', '8')
('5', '3')
('5', '8')
('5', '8')
('5', '10')
('5', '3')
('5', '10')
('5', '10')
('5', '8')
('6', '2')
('5', '8')
('6', '1')
('5', '9')
('5', '6')
('6', '0')
('5', '10')
('5', '9')
('5', '2')
('5', '8')
('5', '8')
('5', '4')
('6', '0')
('6', '0')
('6', '3')
('5', '9')
('6', '5')
('5', '11')
('6', '0')
('6', '2')
('5', '11')
('5', '11')
('5', '9')
('5', '9')
('5', '9')
('5', '10')
('6', '0')
('5', '10')
('5', '8')
('6', '1')
('5', '8')
('5', '7')
('5', '9')
('5', '8')
('5', '9')
('5', '4')
('5', '2')
('5', '5')
('

('6', '2')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '8')
('5', '8')
('5', '8')
('5', '8')
('5', '8')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '9')
('5', '9')
('5', '11')
('5', '7')
('5', '9')
('6', '1')
('6', '1')
('6', '0')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '8')
('5', '8')
('5', '8')
('5', '8')
('5', '7')
('5', '6')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '9')
('5', '5')
('5', '8')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('5', '9')
('5', '9')
('5', '8')
('5', '8')
('5', '8')
('5', '7')
('5', '7')
('5', '7')
('5', '6')
('5', '11.5')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '10

('5', '6')
('6', '0')
('5', '8')
('5', '10')
('5', '10')
('6', '0')
('5', '7')
('5', '9')
('5', '7')
('6', '1')
('6', '1')
('6', '0')
('5', '9')
('5', '11')
('5', '7')
('6', '0')
('6', '0')
('6', '1')
('5', '11')
('5', '8')
('5', '11')
('5', '10')
('6', '1')
('5', '9')
('5', '10')
('5', '10')
('6', '2')
('6', '3')
('6', '1')
('5', '9')
('6', '3')
('5', '6')
('6', '0')
('5', '9')
('6', '0')
('5', '10')
('5', '9')
('5', '11')
('6', '3')
('5', '8')
('5', '11')
('5', '10')
('6', '3')
('6', '1')
('6', '2')
('5', '9')
('6', '0')
('5', '9')
('5', '10')
('6', '3')
('6', '1')
('5', '8')
('5', '9')
('5', '6')
('5', '10')
('5', '9')
('5', '11')
('6', '1')
('5', '7')
('6', '1')
('6', '0')
('5', '8')
('6', '1')
('6', '1')
('6', '0')
('5', '11')
('5', '9')
('5', '8')
('5', '7')
('5', '10')
('5', '9')
('5', '9')
('6', '0')
('5', '9')
('6', '4')
('5', '11')
('6', '2')
('6', '4')
('5', '10')
('6', '0')
('5', '7')
('5', '10')
('5', '9')
('5', '9')
('5', '8')
('6', '2')
('6', '1')
('6', '2')
('6', '0')
(

('5', '7')
('5', '7')
('5', '11')
('5', '6')
('5', '8')
('5', '6')
('5', '10')
('5', '6')
('5', '7')
('6', '3')
('6', '3')
('5', '9')
('6', '0')
('6', '3')
('6', '2')
('6', '2')
('6', '2')
('6', '0')
('6', '4')
('6', '0')
('6', '4')
('6', '0')
('5', '8')
('6', '1')
('5', '7')
('6', '0')
('5', '9')
('6', '2')
('6', '2')
('6', '0')
('5', '9')
('6', '0')
('5', '8')
('6', '2')
('5', '11')
('6', '4')
('5', '11')
('5', '10')
('6', '2')
('6', '0')
('5', '8')
('5', '8')
('5', '10')
('5', '9')
('5', '10')
('5', '5')
('5', '10')
('5', '11')
('6', '2')
('6', '0')
('5', '6')
('5', '9')
('5', '6')
('6', '1')
('5', '10')
('6', '1')
('6', '3')
('6', '1')
('6', '1')
('6', '0')
('5', '7')
('6', '3')
('5', '11')
('5', '11')
('6', '1')
('5', '6')
('6', '5')
('6', '3')
('5', '8')
('6', '4')
('6', '3')
('6', '5')
('5', '11')
('5', '9')
('6', '2')
('6', '2')
('5', '9')
('6', '1')
('5', '10')
('5', '7')
('5', '11')
('5', '9')
('5', '10')
('5', '8')
('5', '10')
('5', '11')
('5', '11')
('6', '2')
('6', '0')
('

('5', '11')
('6', '0')
('6', '2')
('5', '7')
('5', '8')
('5', '9')
('5', '11')
('5', '10')
('5', '10')
('5', '8')
('5', '9')
('6', '0')
('5', '7')
('6', '3')
('5', '9')
('6', '3')
('6', '3')
('6', '1')
('5', '11')
('5', '10')
('5', '9')
('5', '10')
('5', '9')
('5', '11')
('6', '0')
('5', '9')
('6', '0')
('5', '10')
('6', '0')
('5', '11')
('5', '11')
('5', '10')
('6', '3')
('5', '10')
('5', '3')
('5', '10')
('5', '8')
('5', '11')
('6', '0')
('5', '5')
('6', '0')
('6', '1')
('5', '11')
('5', '10')
('5', '11')
('6', '0')
('6', '1')
('6', '3')
('5', '7')
('5', '11')
('5', '10')
('6', '0')
('6', '1')
('5', '8')
('5', '11')
('6', '2')
('6', '3')
('5', '11')
('6', '3')
('6', '2')
('6', '3')
('5', '10')
('6', '0')
('6', '2')
('6', '2')
('6', '1')
('5', '9')
('6', '0')
('5', '0')
('6', '2')
('6', '0')
('6', '1')
('5', '7')
('6', '2')
('5', '11')
('5', '8')
('6', '2')
('5', '9')
('5', '5')
('5', '7')
('5', '10')
('5', '10')
('5', '9')
('6', '1')
('5', '9')
('6', '0')
('6', '0')
('5', '9')
('5', 

('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '1')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('6', '0')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '9')
('5', '8')
('5', '8')
('5', '8')
('5', '7')
('5', '7')
('5', '7')
('5', '7')
('5', '7')
('5', '6')
('5', '5')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '10')
('5', '6')
('5', '4')
('5', '10')
('6', '1')
('5', '11')
('5', '11')
('5', '11')
('5', '10')
('5', '11')
('6', '3')
('6', '3')
('6', '3')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', '2')
('6', 

('6', '5')
('6', '6')
('6', '3')
('5', '11')
('6', '2')
('5', '11')
('5', '9')
('5', '10')
('5', '9')
('5', '8')
('6', '2')
('5', '10')
('5', '6')
('5', '11')
('6', '3')
('6', '0')
('5', '10')
('6', '1')
('6', '2')
('5', '11')
('5', '6')
('5', '11')
('5', '8')
('6', '1')
('5', '8')
('5', '11')
('6', '0')
('6', '0')
('5', '10')
('5', '10')
('5', '7')
('5', '9')
('6', '2')
('6', '2')
('6', '0')
('5', '11')
('5', '10')
('6', '2')
('6', '4')
('6', '2')
('5', '9')
('5', '8')
('6', '4')
('5', '8')
('6', '0')
('5', '10')
('5', '9')
('6', '0')
('6', '5')
('6', '4')
('6', '1')
('6', '0')
('6', '1')
('6', '1')
('5', '9')
('5', '11')
('5', '4')
('5', '10')
('5', '8')
('5', '9')
('5', '8')
('6', '0')
('6', '1')
('5', '10')
('5', '5')
('6', '0')
('6', '2')
('5', '6')
('5', '6')
('5', '11')
('5', '10')
('5', '8')
('5', '9')
('5', '9')
('5', '11')
('5', '5')
('6', '2')
('5', '9')
('6', '2')
('5', '10')
('6', '3')
('6', '3')
('5', '8')
('5', '8')
('5', '7')
('5', '9')
('5', '10')
('5', '10')
('5', '9'

('5', '5')
('5', '9')
('6', '4')
('6', '0')
('5', '11')
('5', '7')
('6', '2')
('6', '5')
('6', '4')
('6', '3')
('5', '9')
('6', '1')
('6', '0')
('5', '11')
('5', '8')
('5', '11')
('6', '0')
('5', '11')
('5', '7')
('5', '10')
('6', '1')
('6', '1')
('6', '0')
('5', '9')
('5', '11')
('5', '8')
('6', '2')
('6', '1')
('6', '2')
('6', '0')
('5', '11')
('6', '1')
('5', '8')
('6', '1')
('5', '10')
('5', '8')
('5', '11')
('5', '8')
('6', '0')
('5', '9')
('6', '0')
('5', '11')
('5', '6')
('5', '9')
('5', '9')
('6', '2')
('6', '3')
('5', '7')
('5', '10')
('5', '11')
('6', '1')
('5', '9')
('6', '1')
('5', '10')
('6', '3')
('5', '10')
('6', '4')
('5', '9')
('5', '7')
('5', '8')
('5', '11')
('5', '8')
('5', '10')
('6', '1')
('5', '8')
('5', '10')
('5', '10')
('5', '4')
('6', '0')
('5', '10')
('6', '0')
('5', '10')
('6', '0')
('5', '10')
('6', '0')
('5', '10')
('6', '0')
('5', '11')
('5', '6')
('6', '0')
('5', '6')
('5', '11')
('5', '10')
('5', '9')
('6', '1')
('5', '8')
('5', '6')
('6', '2')
('6', '

KeyboardInterrupt: 

In [391]:
height_regex = "^(\d)'\s*(\d*\.*\d*)"
def height_str_to_val(height_str):
    match = re.match("^(\d)'\s*(\d*\.*\d*)", height_str)
    groups = match.groups()
    feet, inches = int(groups[0]), float(groups[1])
    return feet * 12 + inches

In [392]:
data['Height'] = data['Height'].apply(height_str_to_val)

DB - Height, Weight, Forty Yard Dash, Shuttle, 3 Cone, Vertical Jump

In [408]:
cb = data[data['Position'] == 'CB']
cb = cb[['Full Name', 'Position', 'Height', 'Weight', 'Forty Yard Dash', 'Shuttle', '3 Cone', 'Vertical Jump']]
cb.head(20)

Full Name Position  Height  Weight  Forty Yard Dash  Shuttle  \
22      Domani Jackson       CB   72.00   181.0            4.510    4.250   
25      Jason Marshall       CB   72.00   167.5            4.590      NaN   
26      Terrion Arnold       CB   71.75   175.0            4.560      NaN   
31   Jaheim Singletary       CB   73.00   172.5            4.620      NaN   
35       Travis Hunter       CB   72.00   165.0              NaN      NaN   
38     William Johnson       CB   74.00   180.0              NaN      NaN   
55    Jakailin Johnson       CB   72.00   170.0            4.500      NaN   
60        Nyland Green       CB   73.00   172.0            4.610      NaN   
76      Jordan Hancock       CB   72.00   158.5            4.630      NaN   
90     Khamauri Rogers       CB   71.00   143.0              NaN      NaN   
108    Daylen Everette       CB   73.00   180.0              NaN      NaN   
156      Sharif Denson       CB   70.00   160.0            4.600      NaN   
165      Jaylen Mbakwe       CB   71.00   170.0              NaN      NaN   
174      Jahlil Hurley       CB   73.00   165.0              NaN      NaN   
198    Earl Little Jr.       CB   71.00   165.5            4.580      NaN   
200    Azareyeh Thomas       CB   73.00   176.5              NaN      NaN   
211      Tony Mitchell       CB   72.00   176.0            4.880    4.560   
223  Marquis Killebrew       CB   72.00   180.0              NaN      NaN   
228    Damarius McGhee       CB   71.50   160.0            4.560    4.250   
231     Shawn Russ Jr.       CB   72.00   168.0            5.027    4.555   

     3 Cone  Vertical Jump  
22      NaN           31.0  
25      NaN            NaN  
26      NaN            NaN  
31      NaN            NaN  
35      NaN            NaN  
38      NaN            NaN  
55      NaN            NaN  
60      NaN           34.1  
76      NaN            NaN  
90      NaN            NaN  
108     NaN            NaN  
156     NaN            NaN  
165     NaN            NaN  
174     NaN            NaN  
198     NaN            NaN  
200     NaN            NaN  
211     NaN           30.8  
223     NaN            NaN  
228     NaN           30.8  
231   7.511           22.2

In [409]:
def create_percentile(position, event):
    """Take in a position - CB, WR, QB, etc. and an event - Fourty Yard Dash, Weight, 3 Cone and return a list of percentile values
        of 0, 10th, ... 100th percentile. Reverse for all speed events (lower is better) such as 40, 3 Cone, and Shuttle."""
    sorted_athletes = data[data['Position'] == position]     
    event_values = (sorted_athletes[event].astype(float)).dropna()
    percentile = []
    for i in np.arange(0, 1.01, 0.01):
        percentile.append(event_values.quantile(i))
    if (event == "Forty Yard Dash") or (event == "Shuttle") or (event == "3 Cone"):
        return percentile[::-1]
    return percentile

Input 1: [89.0, 130.0, 139.0, 145.0, 150.0, 156.5, 160.0, 165.5, 170.0, 180.0, 287.0]
Input 2: 181

181 falls into the 90th percentile -> score: 9

In [410]:
def value_to_score(event, value):
    if (event == "Forty Yard Dash") or (event == "Shuttle") or (event == "3 Cone"):
        return round(((100 - scipy.stats.percentileofscore(metric_dict[event], value, kind='rank')) / 10), 2)
    else:
        return round(((scipy.stats.percentileofscore(metric_dict[event], value, kind='rank')) / 10), 1)

In [412]:
metric_dict = {}
for metric in cb.columns[2:8]:
    metric_dict[metric] = create_percentile('CB', metric)

In [413]:
metric_dict

{'Height': [57.0,
  62.0,
  63.0,
  64.0,
  64.0,
  64.0,
  65.0,
  65.0,
  65.0,
  65.1875,
  66.0,
  66.0,
  66.0,
  66.0,
  66.0,
  66.0,
  66.0,
  67.0,
  67.0,
  67.0,
  67.0,
  67.0,
  67.0,
  67.0,
  67.0,
  67.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  68.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.0,
  69.5,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.0,
  70.5,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.0,
  71.75,
  72.0,
  72.0,
  72.0,
  72.0,
  72.0,
  72.0,
  72.0,
  72.0,
  72.0,
  73.0,
  73.0,
  73.0,
  73.0,
  73.0,
  73.5,
  74.0,
  74.0,
  75.0,
  79.0],
 'Weight': [89.0,
  107.75,
  113.0,
  117.0,
  120.0,
  121.0,
  124.0,
  125.0,
  127.0,
  128.875,
  130.0,
  130.0,
  131.0,
  132.0,
  133.0,
  135.0,
  135.0,
  136.0,
  1

In [414]:
def add_score(metric, data):
    data = data.dropna()
    data[metric + " " + "Score"] = data.apply(lambda x: value_to_score(metric, x[metric]), axis = 1)
    return data

In [416]:
data_new = cb
chosen_metrics = data_new.columns[2:8]
for metric in chosen_metrics:
    data_new = add_score(metric, data_new)

<ipython-input-414-dd2af720b2aa>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[metric + " " + "Score"] = data.apply(lambda x: value_to_score(metric, x[metric]), axis = 1)


In [417]:
data_new.head(20)

Full Name Position  Height  Weight  Forty Yard Dash  Shuttle  \
231      Shawn Russ Jr.       CB   72.00   168.0            5.027    4.555   
296    Antonio Robinson       CB   67.75   151.5            5.097    4.434   
411        Jaden Mickey       CB   66.00   130.0            5.120    4.348   
412     Marcus Scott II       CB   72.00   145.0            4.910    5.900   
497      Omarion Cooper       CB   71.50   168.0            4.967    4.955   
499     Jaylin Marshall       CB   73.00   173.0            4.590    4.873   
534      Trevell Mullen       CB   71.00   145.5            4.794    4.886   
566         Fabian Ross       CB   71.00   193.0            4.660    4.451   
575       Kaleb Higgins       CB   73.25   185.5            4.846    4.218   
634         Ryan Barnes       CB   68.00   136.0            5.044    4.477   
655       Camron Miller       CB   72.00   169.0            4.635    4.373   
683     Adontae Balfour       CB   72.00   148.0            5.053    4.566   
684   Tar'varish Dawson       CB   69.00   149.0            4.765    4.746   
723     Darian Chestnut       CB   69.50   180.0            4.850    4.430   
772      Zakhari Spears       CB   73.00   181.5            4.600    4.507   
820    Jonquis Hardaway       CB   72.00   138.0            4.915    4.482   
944      Jarell Stinson       CB   68.00   139.0            4.560    4.690   
946        Nasir Bowers       CB   68.00   166.0            4.806    4.677   
988        Cameron Epps       CB   73.00   184.5            4.913    4.884   
1037   Darren\t Barkins       CB   71.00   153.0            4.682    4.370   

      3 Cone  Vertical Jump  Height Score  Weight Score  \
231    7.511         22.200           8.6           7.2   
296    7.715         27.500           2.6           4.2   
411    6.978         24.100           1.4           1.1   
412    7.823         23.000           8.6           3.1   
497    7.560         33.500           8.0           7.2   
499    8.728         24.500           9.3           8.2   
534    8.217         24.600           7.5           3.2   
566    7.850         30.000           7.5           9.8   
575    7.769         31.500           9.5           9.6   
634    7.310         24.700           3.3           1.8   
655    7.345         31.012           8.6           7.3   
683    7.368         27.200           8.6           3.5   
684    7.560         18.000           4.6           3.6   
723    7.220         26.500           5.3           9.2   
772    7.538         36.000           9.3           9.4   
820    8.000         22.300           8.6           2.0   
944    7.384         26.000           3.3           2.1   
946    7.502         27.300           3.3           7.0   
988    7.450         27.500           9.3           9.4   
1037   7.162         31.370           7.5           4.4   

      Forty Yard Dash Score  Shuttle Score  3 Cone Score  Vertical Jump Score  
231                    3.27           5.74          7.03                  1.8  
296                    2.57           7.23          5.15                  6.2  
411                    2.38           8.22          9.60                  3.4  
412                    4.36           0.10          4.26                  2.5  
497                    3.86           1.39          6.63                  9.3  
499                    8.51           1.98          0.40                  3.8  
534                    5.94           1.88          1.58                  3.9  
566                    7.33           7.03          3.96                  7.9  
575                    5.25           9.21          4.75                  8.6  
634                    3.07           6.83          8.42                  4.0  
655                    7.52           8.02          8.22                  8.4  
683                    2.97           5.64          8.12                  6.0  
684                    6.14           3.17          6.63                  0.2  
723            

In [418]:
scores = [x + " " + "Score" for x in chosen_metrics]
scores

['Height Score',
 'Weight Score',
 'Forty Yard Dash Score',
 'Shuttle Score',
 '3 Cone Score',
 'Vertical Jump Score']

In [420]:
data_score_only = data_new[["Full Name", "Position"] + scores]
data_score_only.head(20)

Full Name Position  Height Score  Weight Score  \
231      Shawn Russ Jr.       CB           8.6           7.2   
296    Antonio Robinson       CB           2.6           4.2   
411        Jaden Mickey       CB           1.4           1.1   
412     Marcus Scott II       CB           8.6           3.1   
497      Omarion Cooper       CB           8.0           7.2   
499     Jaylin Marshall       CB           9.3           8.2   
534      Trevell Mullen       CB           7.5           3.2   
566         Fabian Ross       CB           7.5           9.8   
575       Kaleb Higgins       CB           9.5           9.6   
634         Ryan Barnes       CB           3.3           1.8   
655       Camron Miller       CB           8.6           7.3   
683     Adontae Balfour       CB           8.6           3.5   
684   Tar'varish Dawson       CB           4.6           3.6   
723     Darian Chestnut       CB           5.3           9.2   
772      Zakhari Spears       CB           9.3           9.4   
820    Jonquis Hardaway       CB           8.6           2.0   
944      Jarell Stinson       CB           3.3           2.1   
946        Nasir Bowers       CB           3.3           7.0   
988        Cameron Epps       CB           9.3           9.4   
1037   Darren\t Barkins       CB           7.5           4.4   

      Forty Yard Dash Score  Shuttle Score  3 Cone Score  Vertical Jump Score  
231                    3.27           5.74          7.03                  1.8  
296                    2.57           7.23          5.15                  6.2  
411                    2.38           8.22          9.60                  3.4  
412                    4.36           0.10          4.26                  2.5  
497                    3.86           1.39          6.63                  9.3  
499                    8.51           1.98          0.40                  3.8  
534                    5.94           1.88          1.58                  3.9  
566                    7.33           7.03          3.96                  7.9  
575                    5.25           9.21          4.75                  8.6  
634                    3.07           6.83          8.42                  4.0  
655                    7.52           8.02          8.22                  8.4  
683                    2.97           5.64          8.12                  6.0  
684                    6.14           3.17          6.63                  0.2  
723                    5.15           7.33          8.81                  5.4  
772                    8.07           6.24          6.83                  9.7  
820                    4.36           6.83          2.87                  1.9  
944                    8.81           3.96          8.02                  5.0  
946                    5.64           4.16          7.03                  6.1  
988                    4.36           1.88          7.52                  6.2  
1037                   7.23           8.02          9.11                  8.5

Generate a score report for all the players at a chosen position with chosen metrics. 

1. Make sure heights are cleaned. Drop NAs?
2. Run the percentile, value_to_score, and add_score functions

In [427]:
type(data['Height'].values[0]) == np.float64

True

In [481]:
if type(data['Height'].values[0]) != np.float64:
    data['Height'] = data['Height'].apply(height_str_to_val)

def create_percentile(position, event):
    """Take in a position - CB, WR, QB, etc. and an event - Fourty Yard Dash, Weight, 3 Cone and return a list of percentile values
        of 0, 10th, ... 100th percentile. Reverse for all speed events (lower is better) such as 40, 3 Cone, and Shuttle."""
    sorted_athletes = data[data['Position'] == position]     
    event_values = (sorted_athletes[event].astype(float)).dropna()
    percentile = []
    for i in np.arange(0, 1.01, 0.01):
        percentile.append(event_values.quantile(i))
    if (event == "Forty Yard Dash") or (event == "Shuttle") or (event == "3 Cone"):
        return percentile[::-1]
    return percentile 

def value_to_score(event, value):
    """Take in an event - Fourty Yard Dash, Weight, etc and player's value and generate player's score for that event using 
    the percentiles for that event for their position."""
    if (event == "Forty Yard Dash") or (event == "Shuttle") or (event == "3 Cone"):
        return round(((100 - scipy.stats.percentileofscore(metric_dict[event], value, kind='rank')) / 10), 2)
    else:
        return round(((scipy.stats.percentileofscore(metric_dict[event], value, kind='rank')) / 10), 2)
    
def add_score(metric, data):
    #data = data.dropna()
    print(data.shape)
    data[metric + " " + "Score"] = data.apply(lambda x: value_to_score(metric, x[metric]), axis = 1)
    return data

def data_representation_for_algorithm(data, position, metric_list):
    #Only work with players at given position
    data = data[data['Position'] == position]
    data.dropna()
    
    #Generate percentiles for each metric at respective position
    metric_dict = {}
    for metric in metric_list:
        metric_dict[metric] = create_percentile(position, metric)
    print(metric_dict['Wingspan'])
    
    #Given percentiles, generate scores for each player's metrics at chosen position. 
    #For example, if player's height is in 82nd percentile for position, the player's score should be rougly 8.2 for Height
    print(data.shape)
    print(round(((scipy.stats.percentileofscore(metric_dict['Wingspan'], 80, kind='rank')) / 10), 2))
    data_scores = data[["Full Name", "Position"] + metric_list]
    print(data_scores.shape)
    for metric in metric_list:
        data_scores = add_score(metric, data_scores)
    
    score_labels = [x + " " + "Score" for x in metric_list]
    data_score_only = data_scores[["Full Name", "Position"] + score_labels]
    return data_score_only

In [454]:
cb_metrics = ["Height", "Weight", "Forty Yard Dash", "Shuttle", "3 Cone"]

In [473]:
data_representation_for_algorithm(data, 'CB', cb_metrics)

dict_keys(['Height', 'Weight', 'Forty Yard Dash', 'Shuttle', '3 Cone'])
(3976, 39)
(3976, 7)
(3976, 7)
(3976, 8)
(3976, 9)


<ipython-input-472-7c654d2386c3>:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[metric + " " + "Score"] = data.apply(lambda x: value_to_score(metric, x[metric]), axis = 1)


(3976, 10)
(3976, 11)


Full Name Position  Height Score  Weight Score  \
22        Domani Jackson       CB          8.61          9.41   
25        Jason Marshall       CB          8.61          7.23   
26        Terrion Arnold       CB          8.12          8.51   
31     Jaheim Singletary       CB          9.31          8.12   
35         Travis Hunter       CB          8.61          6.68   
...                  ...      ...           ...           ...   
71345   Shawn\t Mitchell       CB          0.10          0.10   
71355        Cameron Lee       CB          0.50          0.10   
71356      Isaiah Steele       CB          0.50          0.10   
71360   Conrad\t Levy Jr       CB          0.10          0.10   
71362   Trevion\t Mizell       CB          0.10          0.10   

       Forty Yard Dash Score  Shuttle Score  3 Cone Score  
22                      9.11           9.01           NaN  
25                      8.51            NaN           NaN  
26                      8.81            NaN           NaN  
31                      7.62            NaN           NaN  
35                       NaN            NaN           NaN  
...                      ...            ...           ...  
71345                    NaN            NaN           NaN  
71355                    NaN            NaN           NaN  
71356                    NaN            NaN           NaN  
71360                    NaN            NaN           NaN  
71362                    NaN            NaN           NaN  

[3976 rows x 7 columns]

In [440]:
relevant_zcruit_metrics = ["Height", "Weight", "Wingspan", "Arm Length", "Hand Size", "Forty Yard Dash", "Shuttle", "3 Cone", "Vertical Jump", "Broad Jump", "Powerball Toss"]

In [441]:
data_relevant = data[["Full Name", "Position"] + relevant_zcruit_metrics]
data_relevant.head(5)

Full Name Position  Height  Weight  Wingspan  Arm Length  Hand Size  \
0   Korey Foreman       DE    76.0   248.0       NaN         NaN        NaN   
1  J.T. Tuimoloau       DE    76.0   275.0       NaN         NaN        NaN   
2       JC Latham       OT    78.0   279.0       NaN         NaN        NaN   
3    Emeka Egbuka       WR    72.5   190.0      75.0         NaN       9.00   
4   Dallas Turner       DE    74.5   223.0      83.5       34.25       9.84   

   Forty Yard Dash  Shuttle  3 Cone  Vertical Jump  Broad Jump  Powerball Toss  
0            4.880    4.700   7.758           30.4       104.0            44.5  
1            4.750      NaN     NaN            NaN         NaN             NaN  
2            5.460    4.870     NaN           26.8         NaN             NaN  
3            4.859    4.200   7.083           35.1       112.0            41.0  
4            4.991    4.487   7.752           26.0       115.0             NaN

DE - Height, Weight, Wingspan, Arm Length, Forty Yard Dash, Shuttle, Vertical Jump, Powerball Toss

In [479]:
de_metrics = ["Height", "Weight", "Wingspan", "Arm Length", "Forty Yard Dash", "Shuttle", "Vertical Jump", "Powerball Toss", "3 Cone"]

In [482]:
data_representation_for_algorithm(data, "DE", de_metrics)

[68.0, 70.97, 71.44, 72.0, 72.38, 73.0, 73.32, 73.5, 74.13, 74.25, 74.35, 74.5, 74.5, 74.75, 75.0, 75.0, 75.0, 75.0, 75.0, 75.0, 75.0, 75.185, 75.5, 75.5, 75.5, 76.0, 76.0, 76.0, 76.0, 76.0, 76.0, 76.0, 76.27000000000001, 76.5, 76.5, 76.5, 76.5, 76.5, 76.68, 76.91499999999999, 77.0, 77.0, 77.0, 77.0, 77.0, 77.075, 77.25, 77.5, 77.5, 77.5, 77.5, 77.5, 77.94, 78.0, 78.0, 78.0, 78.0, 78.0, 78.0, 78.0, 78.0, 78.0, 78.0, 78.305, 78.5, 78.5, 78.5, 78.5, 78.98, 79.0, 79.0, 79.0, 79.0, 79.0, 79.0, 79.25, 79.5, 79.5, 79.58, 79.815, 80.0, 80.0, 80.0, 80.0, 80.0, 80.25, 80.5, 80.945, 81.0, 81.0, 81.0, 81.13499999999999, 81.25, 81.5, 81.67999999999999, 82.0, 82.11999999999999, 83.0, 83.06, 84.03, 85.0]
(4229, 39)
8.22
(4229, 11)
(4229, 11)
(4229, 12)
(4229, 13)


<ipython-input-481-0faed75baf67>:27: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[metric + " " + "Score"] = data.apply(lambda x: value_to_score(metric, x[metric]), axis = 1)


KeyError: 'Wingspan'